test/benchmarks/test_experiment_runner.py

```diff
@@ -0,0 +1,46 @@
+import unittest
+
+import subprocess
+
+import experiment_runner
+
+EXPERIMENT_RUNNER_PY = experiment_runner.__file__
+
+
+class ExperimentRunnerTest(unittest.TestCase):
+
+  def test_alexnet_dry_run(self):
+    child = subprocess.run([
+        "python", EXPERIMENT_RUNNER_PY, "--dynamo=openxla", "--dynamo=inductor",
+        "--xla=PJRT", "--xla=None", "--test=eval", "--test=train",
+        "--suite-name=torchbench", "--accelerator=cpu", "--filter=^alexnet$",
+        "--dry-run"
+    ],
+                           capture_output=True,
+                           text=True)
+    expected_in_stderr = [
+        "'--experiment-config={\"accelerator\": \"cpu\", \"xla\": \"PJRT\", \"xla_flags\": null, \"dynamo\": \"openxla\", \"test\": \"eval\"}', '--model-config={\"model_name\": \"alexnet\"}'",
+        "'--experiment-config={\"accelerator\": \"cpu\", \"xla\": \"PJRT\", \"xla_flags\": null, \"dynamo\": \"openxla\", \"test\": \"train\"}', '--model-config={\"model_name\": \"alexnet\"}'",
+    ]
+    for expected in expected_in_stderr:
+      self.assertTrue(expected in child.stderr)
+
+  def test_alexnet_dry_run_cuda(self):
+    child = subprocess.run([
+        "python", EXPERIMENT_RUNNER_PY, "--dynamo=openxla", "--dynamo=inductor",
+        "--xla=PJRT", "--xla=None", "--test=eval", "--test=train",
+        "--suite-name=torchbench", "--accelerator=cuda", "--filter=^alexnet$",
+        "--dry-run"
+    ],
+                           capture_output=True,
+                           text=True)
+    expected_in_stderr = [
+        "'--experiment-config={\"accelerator\": \"cuda\", \"xla\": \"PJRT\", \"xla_flags\": null, \"dynamo\": \"openxla\", \"test\": \"eval\"}', '--model-config={\"model_name\": \"alexnet\"}'",
+        "'--experiment-config={\"accelerator\": \"cuda\", \"xla\": \"PJRT\", \"xla_flags\": null, \"dynamo\": \"openxla\", \"test\": \"train\"}', '--model-config={\"model_name\": \"alexnet\"}'",
+    ]
+    for expected in expected_in_stderr:
+      self.assertTrue(expected in child.stderr)
```

Comment:
This tests if all the configurations in `expected_in_stderr` are in the actual `stderr`. However, it doesn't test the other way around: if `expected_in_stderr` has all the actually generated configurations. I think it would be nice to have that as well.

Reply:
Agreed. Added a check for the number of configs to expect.